src/adapters/redis/index.ts

```diff
@@ -70,14 +70,11 @@ export class RedisStorageAdapter implements StorageAdapter {
    * Use set command to set hash value in radish
    */
   public async set(key: string, value: string, expiresIn?: number): Promise<boolean> {
-    const commands: [KeyType, CommandArgument] = [`${CACHE_PREFIX}:${key}`, value];
-
-    if (expiresIn) {
-      commands.push('PX', expiresIn);
-    }
+    const cacheKey = `${CACHE_PREFIX}:${key}`;
+    const expiryMode =  expiresIn ? 'PX' : undefined;
 
     return Boolean(await withTimeout(
-      this.redisInstance.set(...commands), this.options.operationTimeout));
+      this.redisInstance.set(cacheKey, value, expiryMode, expiresIn), this.options.operationTimeout));
```

Comment:
I am not sure if this is identical to what it was. If “set” works with the arguments object, which is most likely, the result of the last commands to be sent will include “undefined” values, and ioredis will need to process these commands in any case (ioredis will ignore them after a small check btw), which negates this optimization.

Reply:
Previous behavior was returned